### ISSUE TITLE
Heartbeat Interval does not seem to work past 2 minutes

### ISSUE THREAD

**Issue by barretodaniel** [transistorsoft/cordova-background-geolocation-lt#190]
Hello,

We are testing the plugin on iOS and we are using prevent suspend with a heartbeat interval to send locations onHeartbeat. When we set the interval to 2 minutes, it works as expected. However, when we set the interval to 5 minutes, we no longer receive posts to the server.

Are there any options that we may be missing?

below is our config

```js
this.commonOptions = {
                        activityRecogntionInterval: 10000,
                        url: appSettings.API_PATH + 'locations',
                        method: 'POST',
                        autoSync: true,
                        debug: false,
                        desiredAccuracy: 0,
                        distanceFilter: 20,
                        disableElasticity: false,
                        autoSyncThreshold: 0,
                        batchSync: false,
                        stopOnTerminate: false,
                        startOnBoot: true,
                        stopTimeout: 5,
                        heartbeatInterval: 300
                    };
                    this.iosOptions = {
                        activityType: 'OtherNavigation',
                        stationaryRadius: 20,
                        stopDetectionDelay: 0,
                        preventSuspend: true,
                        pausesLocationUpdatesAutomatically: false,
                        useSignificantChangesOnly: false,
                        disableMotionActivityUpdates: true,
                    };
```

**Comment by christocracy** [comment]
Show me Xcode logs of your app running for 5 minutes in preventSuspend mode with the app minimized.

**Comment by orrimer** [comment]
Thanks, will report back with the logs.

**Comment by christocracy** [comment]
It's not recommended to `disableMotionActivityUpdates`.

As of iOS 10, **all** accelerometer-use now requires asking the user for permission so you might as well just enable motion-activity-updates since the data is much better.

Also, be very careful with `pausesLocationUpdatesAutomatically: false` unless you're explicitly turning off location-updates with `setPace(false)`.  If you're not careful, location-services will **NEVER** turn off and you'll kill the battery.

**Comment by orrimer** [comment]
[xcode.txt](https://github.com/transistorsoft/cordova-background-geolocation-lt/files/480567/xcode.txt)

**Comment by orrimer** [comment]
Thats about 15 minutes of activity. First heartbeat works, after that nothing.

**Comment by egoossens** [comment]
Same issue here. Could it be linked to the automatic killing of background tasks?

I'm trying to figure out where and why stale tasks are being created in my code.

**Comment by christocracy** [comment]
@barretodaniel iOS refused to grant more bg-time.

```
2016-09-19 12:25:10.204903 The Retail Group[3928:1111429] ► Location-services: OFF
2016-09-19 12:28:07.290872 The Retail Group[3928:1111429] [TSLocationManager]⚠ FORCE KILL BACKGROUND TASK: 23, remaining time: 4.007281
2016-09-19 12:28:07.292161 The Retail Group[3928:1111429] [TSLocationManager]⚠ FORCE KILL BACKGROUND TASK: 25, remaining time: 4.005934
2016-09-19 12:28:07.293106 The Retail Group[3928:1111429] [TSLocationManager]⚠ FORCE KILL BACKGROUND TASK: 7, remaining time: 4.004981
2016-09-19 12:28:07.294020 The Retail Group[3928:1111429] [TSLocationManager]⚠ FORCE KILL BACKGROUND TASK: 8, remaining time: 4.004061
2016-09-19 12:28:07.294947 The Retail Group[3928:1111429] [TSLocationManager]⚠ FORCE KILL BACKGROUND TASK: 9, remaining time: 4.003137
2016-09-19 12:28:07.295838 The Retail Group[3928:1111429] [TSLocationManager]⚠ FORCE KILL BACKGROUND TASK: 10, remaining time: 4.002238
2016-09-19 12:28:07.296726 The Retail Group[3928:1111429] [TSLocationManager]⚠ FORCE KILL BACKGROUND TASK: 11, remaining time: 4.001350
2016-09-19 12:28:07.297616 The Retail Group[3928:1111429] [TSLocationManager]⚠ FORCE KILL BACKGROUND TASK: 13, remaining time: 4.000459
2016-09-19 12:28:07.298506 The Retail Group[3928:1111429] [TSLocationManager]⚠ FORCE KILL BACKGROUND TASK: 14, remaining time: 3.999569
2016-09-19 12:28:07.299399 The Retail Group[3928:1111429] [TSLocationManager]⚠ FORCE KILL BACKGROUND TASK: 15, remaining time: 3.998676
2016-09-19 12:28:07.300418 The Retail Group[3928:1111429] [TSLocationManager]⚠ FORCE KILL BACKGROUND TASK: 16, remaining time: 3.997660
2016-09-19 12:28:07.301340 The Retail Group[3928:1111429] [TSLocationManager]⚠ FORCE KILL BACKGROUND TASK: 17, remaining time: 3.996735
2016-09-19 12:28:07.302261 The Retail Group[3928:1111429] [TSLocationManager]⚠ FORCE KILL BACKGROUND TASK: 22, remaining time: 3.995815
```

@egoossens The plugin's prevent-suspend mechanism works by "juggling" ios background-tasks.  Every 180s, the plugin briefly turns on location-services to refresh the background-time back to 180s.

I been refactoring some of this behaviour while implementing new `#watchPosition` feature.  I'm still testing.

**Comment by barretodaniel** [comment]
@christocracy since the plugin working properly with 2 minutes as the heartbeat interval, now after we have stopped moving we were still receiving notifications with an interval of 20-30s instead of 2 minutes and then after a few hours of that, the position time was the same for 30 minutes and then all of a sudden changed back to getting unique times with the 20-30s interval. Any ideas on whats happening here?

**Comment by christocracy** [comment]
Logs please.

**Comment by barretodaniel** [comment]
We don't have logs. It was a field test of two days without opening the app
in the foreground.

**Comment by christocracy** [comment]
Boot app in XCode and reproduce behaviour.

**Comment by christocracy** [comment]
Try latest version `2.0.11`.  Should be fixed.

**Comment by barretodaniel** [comment]
Thanks Chris, we're going to run a test this weekend and we'll report back.

**Comment by christocracy** [comment]
Let me know if it works for you.  Please close this issue if so.

**Comment by barretodaniel** [comment]
Okay so, after updating to the latest version and setting heartbeatInterval to 3 minutes this is what happened.

The day we installed, it worked fine. When we were still, the app would only send requests every 3 minutes. After the first switch into and out of in_vehicle activity, our app started receiving location updates approximately every 20-30 seconds for the entirety of the weekend and continuing into today. However, Sunday around noon, our server completely stopped receiving requests. Today we closed the app and opened it back up, and our server started receiving all the locations stored in the db, with the time stamps still showing about every 20-30 seconds.
